saleor/checkout/steps.py

```diff
@@ -227,16 +181,42 @@ def validate(self):
         raise InvalidData()
 
     def forms_are_valid(self):
+        copy_address_form = self.forms['same_billing_as_shipping_address']
+        if copy_address_form.is_valid() and copy_address_form.cleaned_data['shipping_same_as_billing']:
+            shipping_address = self.whole_storage['shipping']['address'].copy()
+            billing_address = {
+                'address': shipping_address}
+            self.whole_storage['billing'] = billing_address
```

Comment:
Use save method for saving data. Assign those values to instance and use them in save method. If `forms_are_valid` is valid `save` will be always executed.